kiwi/storage/subformat/vagrant_base.py

```diff
@@ -222,7 +220,7 @@ def get_additional_vagrant_config_settings(self):
         :return: additional vagrant settings
         :rtype: str
         """
-        pass
+        return ''
 
     def _create_box_metadata(self):
         metadata = self.get_additional_metadata() or {}
```

Comment:
unnecessary `or {}` ? It returns an empty dict now

Reply:
yes any implementation of the function in a child class returns a dict. The abstract interface class however returned None. So the type is broken. The same for the other interface class which is expected to return a string

Reply:
see implementations of vagrant_libvirt.py and vagrant_virtualbox.py for details

Reply:
pythons ```if not``` operator will continue to work in the same way as with the None type before. Thus there was no behavior change and the base class methods are not called in the context of vagrant anyway

Reply:
That's why I thought the change is ok and imho the only way to make static type checks happy. An alternative would be ```->  Optional[...]``` but did not like it that much in this case. Thoughts ?

Reply:
I am fine with this current implementation. I also prefer avoiding `Optional[...]`